packages/dds/tree/src/changeset/toDelta.ts

```diff
@@ -0,0 +1,362 @@
+ * @param changeset - The Changeset to convert
+ * @returns A Delta for applying the changes described in the given Changeset.
+ */
+export function toDelta(changeset: T.Changeset): Delta.Root {
+    return convertPositionedMarks<Delta.OuterMark>(changeset.marks);
+}
+
+function convertPositionedMarks<TMarks>(marks: T.PositionedMarks): Delta.PositionedMarks<TMarks> {
+    const out: Delta.PositionedMarks<Delta.Mark> = [];
+    for (const offsetMark of marks) {
+        const offset = offsetMark.offset || 0;
+        const mark = offsetMark.mark;
+        if (Array.isArray(mark)) {
+            for (const attach of mark) {
+                // Inline into `switch(attach.type)` once we upgrade to TS 4.7
+                const type = attach.type;
+                switch (type) {
+                    case "Insert": {
+                        const insertMark: Delta.Insert = {
+                            type: Delta.MarkType.Insert,
+                            content: cloneTreeContent(attach.content),
+                        };
+                        out.push({ offset, mark: insertMark });
+                        break;
+                    }
+                    case "MInsert": {
+                        const clone = cloneAndModify(attach);
+                        if (clone.fields.size > 0) {
+                            const insertMark: Delta.InsertAndModify = {
+                                type: Delta.MarkType.InsertAndModify,
+                                ...clone,
+                            };
+                            out.push({ offset, mark: insertMark });
+                        } else {
+                            const insertMark: Delta.Insert = {
+                                type: Delta.MarkType.Insert,
+                                content: [clone.content],
+                            };
+                            out.push({ offset, mark: insertMark });
+                        }
+                        break;
+                    }
+                    case "MoveIn": break;
+                    case "Bounce":
+                    case "Intake":
+                        fail(ERR_NOT_IMPLEMENTED);
+                        break;
+                    default: unreachableCase(type);
+                }
+            }
+        } else {
+            // Inline into `switch(mark.type)` once we upgrade to TS 4.7
+            const type = mark.type;
+            switch (type) {
+                case "Modify": {
+                    if (mark.tomb === undefined) {
+                        out.push({
+                            offset,
+                            mark: {
+                                type: Delta.MarkType.Modify,
+                                ...convertModify<Delta.OuterMark>(mark),
+                            },
+                        });
+                    }
+                    break;
+                }
+                case "Delete": {
+                    if ("count" in mark) {
```

Comment:
This check makes me feel like DeleteAndModify should have a separate type.

Reply:
Indeed. I started making the change for insert and forgot to do the others.